crates/hir/src/display.rs

```diff
@@ -562,6 +591,16 @@ fn write_where_clause(
 ) -> Result<bool, HirDisplayError> {
     let params = f.db.generic_params(def);
 
+    let container = match def {
```

Comment:
Not a fan of having this here as it clutters the function a bit more. Since we already return whether we wrote any predicates or not via the bool return, can we instead have the extra logic in the caller of this? I'd rather have the `where` part emission duplicated to the caller if nothing got printed than making this function here more complicated I think